src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -149,47 +151,57 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                 PrintProviders(providerCollection, enabledBy);
 
                 DiagnosticsClient diagnosticsClient;
-                Process process;
+                Process process = null;
                 DiagnosticsClientBuilder builder = new DiagnosticsClientBuilder("dotnet-trace", 10);
-                bool shouldResumeRuntime = ProcessLauncher.Launcher.HasChildProc || !string.IsNullOrEmpty(diagnosticPort) || resumeRuntime;
+                bool shouldResumeRuntime = ProcessLauncher.Launcher.HasChildProc || portConfig.IsListenConfig || resumeRuntime;
```

Comment:
If someone explicitly writes --resume-runtime: false on the command-line I think they would be very confused if the tool still sends the resume command. I think we should either honor the false setting or give an error saying that resumeRuntime:false is not allowed in a given situation.

Reply:
Jupp though about that as well, we now potentially have some arguments that would even out each other. I think we should honor --resume-runtime: false in cases where we run child process or is setup in reverse connect scenario, since that is what the user explicitly asked.

Reply:
I implemented that approach in this PR:
shouldResumeRuntime = (ProcessLauncher.Launcher.HasChildProc || portConfig.IsListenConfig) && resumeRuntime;
OK?

Reply:
From our previous discussion didn't we want to send the resume command when connecting? I thought we wanted support scenarios like this:
```
# launch app
export DOTNET_DiagnosticPorts=foo.sock,listen,suspend
myapp.exe

# launch tool and resume
dotnet-trace --diagnostic-port foo.sock,connect ...
```

I was thinking the policy could be really simple:
```
shouldResumeRuntime = resumeRuntime;
```

Reply:
Agree, lets get rid of the complete expression, intent should be to always resume runtime, regardless if we listen,connect and user can override by passing --resume-runtime: false, regardless how you are running.